src/main/scala/firrtl/Emitter.scala

```diff
@@ -563,17 +563,24 @@ class VerilogEmitter extends SeqTransform with Emitter {
             Seq(tabs, "end else ", _falsex.head.tail) +: _falsex.tail
           }
 
-          /* For a Mux assignment, there are five possibilities:
+          /* For a Mux assignment, there are five possibilities, with one subcase for asynchronous reset:
            *   1. Both the true and false condition are self-assignments; do nothing
            *   2. The true condition is a self-assignment; invert the false condition and use that only
-           *   3. The false condition is a self-assignment; skip the false condition
+           *   3. The false condition is a self-assignment
+           *      a) The reset is asynchronous; emit both 'if' and 'else' to avoid latches
+           *      b) The reset is synchronous; skip the false condition
            *   4. The false condition is a Mux; use the true condition and use 'else if' for the false condition
            *   5. Default; use both the true and false conditions
            */
           (m.tval, m.fval) match {
             case (t, f) if weq(t, r) && weq(f, r) => Nil
             case (t, _) if weq(t, r)              =>  _ifNot +: _false                           :+ _end
-            case (_, f) if weq(f, r)              =>  _if    +: _true                            :+ _end
+            case (_, f) if weq(f, r) =>
+              if (m.cond.tpe == AsyncResetType) {
+                (_if +: _true :+ _else) ++ _false :+ _end     // if async reset (case 3a) avoid latches
+              } else {
+                _if +: _true :+ _end                          // else: case 3b
+              }
```

Comment:
I tried like 10 ways to nest statements to try to minimally mess up the pretty formatting, but I didn't think of that option, which is clearly better :)

Reply:
I changed it to that style. I think the match statement starting on the same line right after the guard might break up the visual flow, so I could also just curly brace deeper, but that would break it up more vertically.